### ISSUE TITLE
Use log instead of println!/eprintln!

### ISSUE THREAD

**Issue by andreeaflorescu** [rust-vmm/vmm-reference#79]
Import the `log` crate and use it to log needed information.

For now it should be enough to just replace the existing `println!` or `eprintln!`.

**Comment by tumbleshack** [comment]
Hey! 

My friend @seanmichwalsh and I are looking to learn about VVM's and Rust. Mind if we take on this issue? 

It may take us a bit to learn our way around if that's okay with you.

**Comment by andreeaflorescu** [comment]
Sure @tumbleshack , sounds good. If you need any help, let us know.

You can also find us on Slack: https://join.slack.com/t/rust-vmm/shared_invite/enQtODAxMzA2ODIyMTc2LWRhYjIwZmQ0YzUxODJlMTRhZWU2ZDBjYmJiNzBmOWVmYjg4MjY5YWRjYjM0YzQ5YzgyMTBmYzNlMjMzYmZlODU

**Comment by dodsonmg** [comment]
Is this still open?  I'm happy to pick up where #114 left off...

**Comment by andreeaflorescu** [comment]
@dodsonmg yes, this is still opened. We should though reduce the scope for the beginning so that's easier to get it merged. We can do this initially for the `main.rs`, and the `vmm` and `api` crates.

**Comment by dodsonmg** [comment]
Cool.  Thanks @andreeaflorescu.  I'll pick it up.

**Comment by dodsonmg** [comment]
Do you want the log level and log file to be command line arguments, or do you just want default log level settings and log output to go to stderr?

If the former, the log level and file can't be established until we've processed command line arguments, and we'll miss some of the `println!` and `eprintln!` statements that come before then.  This is how cloud hypervisor does it.

If the latter, then I think we can just initialise the logger first and sweep everything up.

**Comment by andreeaflorescu** [comment]
I think how it worked in Firecracker is that we were using `println` and `eprintln` until the logger is initialized, and `error` `info` afterwards. This way you don't loose any logs because you're redirecting the output to a file anyway. I think we could start with something simple where we assume stdout and stderr because I believe these can also just directly be forwarded to files without us doing anything special in the VMM code.
